frappe/core/doctype/user/user.py

```diff
@@ -698,28 +706,19 @@ def has_email_account(email):
 
 @frappe.whitelist(allow_guest=False)
 def get_email_awaiting(user):
-	waiting = frappe.db.sql("""select email_account,email_id
-		from `tabUser Email`
-		where awaiting_password = 1
-		and parent = %(user)s""", {"user":user}, as_dict=1)
+	waiting = frappe.get_all("User Email", fields=["email_account", "email_id"], filters={"awaiting_password": 1, "parent": user})
 	if waiting:
 		return waiting
 	else:
-		frappe.db.sql("""update `tabUser Email`
-				set awaiting_password =0
-				where parent = %(user)s""",{"user":user})
+		user_email_table = DocType("User Email")
+		frappe.qb.update(user_email_table).set(user_email_table.user_email_table, 0).where(user_email_table.parent == user)
 		return False
 
 def ask_pass_update():
 	# update the sys defaults as to awaiting users
 	from frappe.utils import set_default
 
-	doctype = DocType("User Email")
-	users = frappe.qb.from_(doctype).where(doctype.awaiting_password == 1).select(
-		doctype.parent.as_("user")
-	).distinct().run(as_dict=True)
-
-	password_list = [ user.get("user") for user in users ]
+	password_list = frappe.get_all("User Email", filters={"awaiting_password": 1}, distinct=True, pluck="parent")
```

Comment:
It's better to use bool values for fields of type check. Use int for Int types only perhaps? Again, doesn't affect the outcome tho.

```suggestion
	password_list = frappe.get_all("User Email", filters={"awaiting_password": True}, pluck="parent", distinct=True)
```